### ISSUE TITLE
The Symmetry of the Metric

### ISSUE THREAD

**Issue by ischencheng** [oooolga/JEDi#4]
I found the JEDi isn't symmetric on a video prediction task. Swaping the train_loader and test_loader produced different results.

`jedi.load_features(train_loader=generated_dl, test_loader=unpaired_target_dl, num_samples=256)`

When the train_loader=generated_dl, test_loader=unpaired_target_dl, jedi=1.28. When the train_loader=unpaired_target_dl, test_loader=generated_dl, jedi=1.39.

I am wondering the exact meaning of train_loader and test_loader.

**Comment by oooolga** [comment]
Hi @ischencheng,

The code should produce the same metric values when using the same set of training/testing features, even if their roles are swapped.

For example, the following code swapped `train_features` and `test_features` in place:
```
from videojedi import JEDiMetric
import numpy as np
jedi = JEDiMetric()
jedi.train_features = np.random.rand(5000, 1280)
jedi.test_features = np.random.rand(5000, 1280)
print(f"JEDi Metric: {jedi.compute_metric()}")
temp = jedi.train_features.copy()
jedi.train_features = jedi.test_features.copy()
jedi.test_features = temp
print(f"JEDi Metric: {jedi.compute_metric()}")
```
and the output is:
```
JEDi Metric: 0.0020221921892465966
JEDi Metric: 0.0020221921892465966
```

It seems your code might be using different sets of training/testing features when computing the metric. Possible reasons could include:
1. Your data loader has shuffle=True.
2. You need to reset your data loader to ensure you are comparing the same set of examples.

Let me know if my explanation resolves your confusion. If the metric still computes different values, feel free to share more details.
